docs/how_to/resource_tracking.md

```diff
@@ -0,0 +1,157 @@
+
+The difference between **total produced** and **total consumed** gives the amount of **available resources** for a particular `resource pool attribute`.
+
+![consumption](../images/consumption.png)
+
+### Resource group
+
+A `resource group` is a definition of a resource of same kind, composed of attributes that may **produce** or 
+**consume** from a particular `resource pool`.
+
+Example of resource group:
+
+- Bare metal servers
+- Physical disks
+- VMware VMs
+- K8S namespaces
+- Openshift projects
+- Openstack tenants
+
+### Resource
+
+A resource is an instance of a `resource group` definition.
+Resources can be created, updated or deleted from the Squest UI or API.
+Updating resources in a resource group impact the total amount of produced or consumed resource on pool's attributes.
+
+## Multiple layer example
+
+In this example we do track the consumption of an orchestrator of container like Kubernetes or Openshift. 
+Namespaces (or projects in Openshift world) are a way to divide cluster resources between multiple users by using 
+resource quota.
+
+Openshift and Kubernetes frameworks are commonly deployed in a virtual machines. 
+So we retrieve layers from previous example with bare metal servers that produce resources in an hypervisor.
+
+Orchestrators are usually composed of 2 kind of node: Masters and Workers.
+Master VMs are used by the infrastructure itself and workers for user's workloads, aka namespaces.
+
+As namespaces are only executed in "worker" nodes, we need to declare 2 different **resource group**: 
+
+- Master VMs 
+- Worker VMs 
+
+The aggregation of resources of all workers compose the **resource pool** of available resources that the `namespaces` 
+**resource group** will consume.
+
+The complete resource tracking definition would look like the following:
+
+![openshift_example](../images/openshift_example.png)
+
+With this definition, we are able to determine there is enough available resources in pools to handle underlying objects.
+
+Adding a new namespace in the last **resource group** `namespaces` will generate more consumption on the `K8S` 
+**resource pool**. If this last pool is lacking of resources, adding more worker node in the `worker VMs` **resource 
+group** will be required, generating consumption on upper layers and so on...
+
+## Link service catalog instances to resources
+
+Resources can be created from the API. It allows to create automatically a new resource in a resource group when 
+something is provisioned from the service catalog.
+
+In the example below, the playbook executed in Tower/AWX would have created a VM. 
+At the end of the process we call the squest API to instantiate a resource in the right resource group to reflect the 
+consumption.
+We link as well the pending instance(given by `squest.instance.id`) to this resource via the flag `service_catalog_instance`.
+```yaml
```

Comment:
Can you use the yaml file (playbook_examples/add_resource_in_resource_group.yml) instead of text?

Reply:
I tried with the markdown extension "pymdownx.snippets" but the result is very bad